executables/bal.go

```diff
@@ -0,0 +1,306 @@
+		} else {
+			javaCmd = "java"
+		}
+	}
+
+	javaCmd = strings.TrimSpace(javaCmd)
+	var ballerinaCLIWidth string
+
+	// Check if tput is available
+	if !commandExists("tput") {
+		ballerinaCLIWidth = "80"
+	} else {
+		if cols, err := getTerminalColumns(); err != nil {
+			ballerinaCLIWidth = "80"
+		} else {
+			ballerinaCLIWidth = strconv.Itoa(cols)
+		}
+	}
+	os.Setenv("BALLERINA_CLI_WIDTH", ballerinaCLIWidth)
+
+	//Setting Ballerina debug port
+
+	balJavaDebug, isSet := os.LookupEnv("BAL_JAVA_DEBUG")
+	javaOpts := ""
+	if isSet {
+		if balJavaDebug == "" {
+			fmt.Println("Please specify the debug port for the BAL_JAVA_DEBUG variable")
+			os.Exit(1)
+		} else {
+			javaOpts = os.Getenv("JAVA_OPTS")
+
+			if javaOpts != "" {
+				fmt.Println("Warning !!! User specified JAVA_OPTS may interfere with BAL_JAVA_DEBUG")
+			}
+
+			balDebugOpts := os.Getenv("BAL_DEBUG_OPTS")
+
+			if balDebugOpts != "" {
+				javaOpts = javaOpts + " " + balDebugOpts
+			} else {
+				javaOpts = javaOpts + " -Xdebug -Xnoagent -Djava.compiler=NONE -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=" + balJavaDebug
+			}
+		}
+	}
+	jarPath := filepath.Join(ballerinaHome, "bre", "lib")
+	//Ballerina Classpath Setting
+	var ballerinaClasspath string
+	toolsJarPath := filepath.Join(ballerinaHome, "bre", "lib", "tools.jar")
+	if _, err := os.Stat(toolsJarPath); err == nil {
+		ballerinaClasspath = filepath.Join(javaHome, "lib", "tools.jar")
+	}
+
+	breFiles, err := filepath.Glob(filepath.Join(jarPath, "*.jar"))
+	if err == nil {
+		for _, f := range breFiles {
+			ballerinaClasspath += string(filepath.ListSeparator) + f
+		}
+	}
+
+	// Add JAR files from lib/tools/lang-server/lib
+	langServerPath := filepath.Join(ballerinaHome, "lib", "tools", "lang-server", "lib")
+	langServerFiles, err := filepath.Glob(filepath.Join(langServerPath, "*.jar"))
+	if err == nil {
+		for _, f := range langServerFiles {
+			ballerinaClasspath += string(filepath.ListSeparator) + f
+		}
+	}
+
+	// Add JAR files from lib/tools/debug-adapter/lib
+	debugAdapterPath := filepath.Join(ballerinaHome, "lib", "tools", "debug-adapter", "lib")
+	debugAdapterFiles, err := filepath.Glob(filepath.Join(debugAdapterPath, "*.jar"))
+	if err == nil {
+		for _, f := range debugAdapterFiles {
+			ballerinaClasspath += string(filepath.ListSeparator) + f
+		}
+	}
+
+	//Define Ballerina CLI Arguments
+	cmdLineArgs := []string{
+		"-Xms256m", "-Xmx1024m",
+		"-XX:+HeapDumpOnOutOfMemoryError",
+		"-classpath", ballerinaClasspath,
+		"-Dballerina.home=" + ballerinaHome,
+		"-Dballerina.version=" + ballerinaVersion,
+		"-Denable.nonblocking=false",
+		"-Djava.security.egd=file:/dev/./urandom",
+		"-Dfile.encoding=UTF8",
+		"-Dballerina.target=jvm",
+		"-Djava.command=" + javaCmd,
+	}
+
+	if javaOpts != "" {
+		JAVA_OPTS := strings.Fields(javaOpts)
+		cmdLineArgs = append(cmdLineArgs, JAVA_OPTS...)
+	}
+	//Implementation of bal run <*.jar>
+	if len(os.Args) >= 3 && os.Args[1] == "run" && isJarFile(os.Args[2]) {
```

Comment:
A person who reads the code might not understand what the values 3, 1, 2 stand for. Let's make them constants and name them so that their purpose is obvious.